VERSIONING.md

```diff
@@ -0,0 +1,90 @@
+# Versioning
+
+This document describes the versioning policy for this repository. This policy
+is designed so that the following goal can be achieved:
+
+**Users are provided a codebase of value that is stable and secure.**
+
+## Policy
+
+* Versioning of this project will be idiomatic of a Go project using [Go
+  modules](https://github.com/golang/go/wiki/Modules).
+    * [Semantic import
+      versioning](https://github.com/golang/go/wiki/Modules#semantic-import-versioning)
+      will be used.
+        * Versions will comply with [semver 2.0](https://semver.org/spec/v2.0.0.html).
+        * If a module is version `v2` or higher, the major version of the module
+          must be included as a `/vN` at the end of the module paths used in
+          `go.mod` files (e.g., `module go.opentelemetry.io/collector/v2`, `require
+          go.opentelemetry.io/collector/v2 v2.0.1`) and in the package import path
+          (e.g., `import "go.opentelemetry.io/collector/v2/component"`). This includes the
+          paths used in `go get` commands (e.g., `go get
+          go.opentelemetry.io/collector/v2@v2.0.1`.  Note there is both a `/v2` and a
+          `@v2.0.1` in that example. One way to think about it is that the module
+          name now includes the `/v2`, so include `/v2` whenever you are using the
+          module name).
+        * If a module is version `v0` or `v1`, do not include the major version in
+          either the module path or the import path.
+    * A single module should exist, rooted at the top level of this repository,
+      that contains all packages provided for use outside this repository.
+    * Additional modules may be created in this repository to provide for
+      isolation of build-time tools or other commands. Such modules should be
+      versioned in sync with the `go.opentelemetry.io/collector` module.
+    * Configuration structures should be considered part of the public API and backward
```

Comment:
Does this mean also the tags in structs?

Reply:
Probably.  https://cs.opensource.google/go/x/exp/+/master:apidiff/README.md doesn't discuss struct tags when talking about compatibility, but since our primary use for configuration structures is to enable defining them in YAML or JSON, we should probably avoid changes to struct tags that would break unmarshalling existing serialized configuration.

Do you think that it is necessary to explicitly call out struct tags here?  If so, do you have a proposal for language to use?  I think the existing language regarding maintaining backward compatibility, along with the single carve-out for additive changes, is sufficient.

Reply:
We do use structtags for configuration (yaml). @Aneurysm9 I would call out explicitly if the yaml configuration is or not consider part of the public APIs guarantees.